src/daemon.rs

```diff
@@ -131,38 +123,112 @@ impl<'a> Daemon<'a> {
         self
     }
 
+    /// Determines where standard output will be piped to since daemons have no console attached
+    ///
+    /// It's highly recommended to set this to a file if you want to see output.
     pub fn stdout<T: Into<Stdio>>(mut self, stdio: T) -> Self {
         self.stdout = stdio.into();
         self
     }
 
+    /// Determines where standard error will be piped to since daemons have no console attached
+    ///
+    /// It's highly recommended to set this to a file if you want to see output.
     pub fn stderr<T: Into<Stdio>>(mut self, stdio: T) -> Self {
         self.stderr = stdio.into();
         self
     }
 
+    /// Set the daemon process name
+    ///
+    /// For example, this is what shows up in `ps`.
     pub fn name(mut self, name: &OsStr) -> Self {
         self.name = Some(OsString::from(name));
         self
     }
 
+    /// Hook called before the fork with the current pid as argument
     pub fn setup_pre_fork_hook(mut self, pre_fork_hook: fn(pid: i32)) -> Self {
         self.before_fork_hook = Some(pre_fork_hook);
         self
     }
 
-    pub fn setup_post_fork_parent_hook(mut self, post_fork_parent_hook: fn(parent_pid: i32, child_pid: i32) -> !) -> Self {
+    /// Hook called after the fork with the parent pid as argument
+    ///
+    /// Can be used to continue some work on the parent after the fork.
+    /// **NOTE:** This hook must call `std::process::exit()`.
```

Comment:
It doesn't **need** to call `exit()`, it can call anything really, the main point is it's not expected to return ever thus the return type being `!`, so as long as it never returns the compiler will be happy.

Reply:
Alright, how's this?